packages/theme-check-common/src/checks/static-stylesheet-and-javascript-tags/index.ts

```diff
@@ -0,0 +1,48 @@
+import { NodeTypes } from '@shopify/liquid-html-parser';
+import { LiquidCheckDefinition, Severity, SourceCodeType } from '../../types';
+
+export const StaticStylesheetAndJavascriptTags: LiquidCheckDefinition = {
+  meta: {
+    code: 'StaticStylesheetAndJavascriptTags',
+    name: 'Reports non static stylesheet and javascript tags',
+    docs: {
+      description:
+        'Reports the usage of Liquid within {% stylesheet %} and {% javascript %} tags, which should only contain static content.',
+      recommended: true,
+      url: 'https://shopify.dev/docs/storefronts/themes/tools/theme-check/checks/static-stylesheet-and-javascript-tags',
+    },
+    type: SourceCodeType.LiquidHtml,
+    severity: Severity.ERROR,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    return {
+      async LiquidRawTag(node) {
+        if (node.name !== 'stylesheet' && node.name !== 'javascript') {
+          return;
+        }
+
+        const liquidNodes = node.body.nodes.filter(
+          (childNode) =>
+            childNode.type === NodeTypes.LiquidVariableOutput ||
+            childNode.type === NodeTypes.LiquidTag ||
+            childNode.type === NodeTypes.LiquidRawTag,
+        );
```

Comment:
We should use a visitor here since you could have child nodes of LiquidTag theoretically, no? This only looks at the top level? 

Maybe we are doing this backwards? Maybe we should check LiquidVariableOutput's _ancestors_ and maybe find a stylesheet/js raw tag in there?

```
async LiquidTag(node, ancestors) {
  const invalidParentNode = ancestors.find((parent) => parent.type === NodeTypes.LiquidRawTag && ['stylesheet', 'javascript'].includes(parent.name));
  if (invalidParentNode) {
     // report
  }
  }
  ```

Reply:
I mean the read squiggly should appear on the `{{ variant }}` variable output here:
```
{% stylesheet %}
  {% if cond %}
     .class--{{ variant }} { color: blue }
  {% endif %}
{% endstylesheet %}
```

The variant wouldn't be a direct child of stylesheet, it would be a child of the if